circuit/src/utils.rs

```diff
@@ -26,7 +26,104 @@ use halo2::{
 };
 use num_bigint::BigUint;
 use rand::Rng;
-use std::{fmt::Debug, fs::write, io::Read, time::Instant};
+use serde::{de::DeserializeOwned, Serialize};
+use std::{
+	env::current_dir,
+	fmt::Debug,
+	fs::{write, File},
+	io::{BufReader, Error as IoError, Read},
+	path::Path,
+	time::Instant,
+};
+
+/// Reads raw bytes from the file
+pub fn read_bytes(path: impl AsRef<Path>) -> Vec<u8> {
+	let f = File::open(path).unwrap();
+	let mut reader = BufReader::new(f);
+	let mut buffer = Vec::new();
+
+	// Read file into vector.
+	reader.read_to_end(&mut buffer).unwrap();
+
+	buffer
+}
+
+/// Reads string from the file
+pub fn read_string(path: impl AsRef<Path>) -> String {
+	let f = File::open(path).unwrap();
+	let mut reader = BufReader::new(f);
+	let mut buffer = String::new();
+
+	// Read file into string.
+	reader.read_to_string(&mut buffer).unwrap();
+
+	buffer
+}
+
+/// Reads raw bytes from the file
+pub fn read_bytes_data(name: &str) -> Vec<u8> {
```

Comment:
How about `bin_file_name` or `file_name`, instead of `name`?

Reply:
Sure, we can do that